KEYCLOAK-18852 Prevent NPE in case of missing truststore

Robbilie opened the pull request:
even though the "return null" at the top of the method is called if no truststore is set, the finally block is still executed. And since the keystore is not there an NPE is thrown when calling the remove method.

Robbilie committed:
KEYCLOAK-18852 Prevent NPE in case of missing truststore

even though the "return null" at the top of the method is called if no truststore is set, the finally block is still executed. And since the keystore is not there an NPE is thrown when calling the remove method.

Robbilie commented:
this whole file has its indentation all over the place :/

actual change is these few lines:
https://github.com/keycloak/keycloak/pull/8282/files#diff-92f474af4048d16450214acadee2d91e7b66b69863bf2c2c782b2c86d2578b0fR211-R214

Robbilie commented:
```
16:24:45,754 WARN  [org.keycloak.services.x509.NginxProxySslClientCertificateLookup] (default task-56) Keycloak Truststore is null, but it is required !
16:24:45,754 WARN  [org.keycloak.services.x509.NginxProxySslClientCertificateLookup] (default task-56)   see https://www.keycloak.org/docs/latest/server_installation/index.html#_truststore
16:24:45,754 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (default task-56) Uncaught server error: java.lang.NullPointerException
	at org.keycloak.keycloak-services@14.0.0//org.keycloak.services.x509.NginxProxySslClientCertificateLookup.buildChain(NginxProxySslClientCertificateLookup.java:211)

```

mposolda reviewed (changes requested):
@Robbilie Thanks for the PR! Few points:
- Could you please create JIRA for this issue? See contributor guidelines https://www.keycloak.org/community and referenced links from it for more details. Also see other commits in the Keycloak repository for the "inspiration"
- Could you please squash the commits in this PR and make sure that there is JIRA ID at the start of the commit message? Again, see other commits for the inspiration.
- Regarding indentation, we try to use 4 spaces and no tabs. I can see that the file NginxProxySslClientCertificateLookup is a bit broken, but I am not 100% sure if it is better after your PR. Could you please try to fix the indentation and make sure there are no tabs and there is 4 spaces indentation in whole file? Again, see other files in workspace for the inspiration. Thanks

Robbilie commented:
@mposolda changed the indentation, did some cleanup, squashed and renamed commit, created ticket and put the ticket number in the commit message at the start

mposolda reviewed (approved):
@Robbilie Thanks!

Robbilie commented:
just out of curiosity, who is merging this then? :)
